### ISSUE TITLE
Transition is not working correctly for light.toggle

### ISSUE THREAD

**Issue by ivlis** [emontnemery/hatasmota#63]
When light is transitioning from OFF to ON fade is enabled, but for ON to OFF is disabled.

Home assistant action:
```yaml
    - entity_id: light.kitchen_table
      service: light.toggle
      data:
        brightness: 255
        color_temp: 500
        transition: 2
```

Tamsota logs:
```
16:35:01 MQT: stat/lohas_light10/POWER = OFF
16:35:07 MQT: stat/lohas_light10/RESULT = {"Fade":"ON"}
16:35:07 MQT: stat/lohas_light10/RESULT = {"Speed":4}
16:35:07 MQT: stat/lohas_light10/RESULT = {"POWER":"ON","Dimmer":100,"Color":"00000000FF","HSBColor":"294,100,0","White":100,"CT":500,"Channel":[0,0,0,0,100]}
16:35:07 MQT: stat/lohas_light10/RESULT = {"POWER":"ON","Dimmer":100,"Color":"00000000FF","HSBColor":"294,100,0","White":100,"CT":500,"Channel":[0,0,0,0,100]}
16:35:12 MQT: stat/lohas_light10/RESULT = {"Fade":"OFF"}
16:35:12 MQT: stat/lohas_light10/RESULT = {"POWER":"OFF"}
16:35:12 MQT: stat/lohas_light10/POWER = OFF
```
For some reason, FADE is OFF.

Also, tasmota supports relative fade now (SetOption117), so this is probably no longer needed: https://github.com/emontnemery/hatasmota/blob/8138a508b1b29439de1abc6c055a29891a1ea264/hatasmota/light.py#L318

**Comment by emontnemery** [comment]
Please enable debug log in HA and provide a log showing what happens when you toggle the light:

```yaml
logger:
  default: info
  logs:
    homeassistant.core: debug
    homeassistant.components.mqtt: debug
```

**Comment by emontnemery** [comment]
Thanks for pointing to SetOption117, this actually makes it worse because hatasmota must support both cases..

**Comment by ivlis** [comment]
HA version: 2021.1.0


Here is the portion of the relevant log:

```
2021-01-07 12:24:29 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=light, service=toggle, service_data=brightness=255, color_temp=500, transition=2, entity_id=['light.kitchen_table']>
2021-01-07 12:24:29 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event call_service[L]: domain=mqtt, service=publish, service_data=topic=cmnd/lohas_light10/Backlog, payload=NoDelay;Fade 0;NoDelay;Power1 OFF>
2021-01-07 12:24:29 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=automation.toggle_kitchen_lights_on_button, old_state=<state automation.toggle_kitchen_lights_on_button=on; last_triggered=2021-01-07T12:24:29.317051-05:00, mode=queued, current=1, max=10, id=e6145af7001f4ad89da61ece88a3b840, friendly_name=Toggle Kitchen Lights on Button @ 2021-01-07T12:23:18.114341-05:00>, new_state=<state automation.toggle_kitchen_lights_on_button=on; last_triggered=2021-01-07T12:24:29.317051-05:00, mode=queued, current=0, max=10, id=e6145af7001f4ad89da61ece88a3b840, friendly_name=Toggle Kitchen Lights on Button @ 2021-01-07T12:23:18.114341-05:00>>
2021-01-07 12:24:29 DEBUG (MainThread) [homeassistant.components.mqtt] Transmitting message on cmnd/lohas_light10/Backlog: 'NoDelay;Fade 0;NoDelay;Power1 OFF', mid: 315
2021-01-07 12:24:29 DEBUG (MainThread) [homeassistant.components.mqtt] Received message on stat/lohas_light10/RESULT: b'{"Fade":"OFF"}'
2021-01-07 12:24:29 DEBUG (MainThread) [homeassistant.components.mqtt] Received message on stat/lohas_light10/RESULT: b'{"POWER":"OFF"}'
2021-01-07 12:24:29 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event state_changed[L]: entity_id=light.kitchen_table, old_state=<state light.kitchen_table=on; min_mireds=153, max_mireds=500, effect_list=['None', 'Wake up', 'Cycle up', 'Cycle down', 'Random'], brightness=255.0, color_temp=500, white_value=255.0, effect=None, friendly_name=Kitchen Table, supported_features=183 @ 2021-01-07T12:23:45.930777-05:00>, new_state=<state light.kitchen_table=off; min_mireds=153, max_mireds=500, effect_list=['None', 'Wake up', 'Cycle up', 'Cycle down', 'Random'], friendly_name=Kitchen Table, supported_features=183 @ 2021-01-07T12:24:29.409528-05:00>>
```

**Comment by emontnemery** [comment]
Thanks!
This is caused by a bug in Home Assistant, and will be fixed by home-assistant/core#44950, which is tagged for next point release.
